packages/apps/human-app/frontend/src/api/services/worker/my-jobs-data.ts

```diff
@@ -48,20 +50,49 @@ const getMyJobsTableData = async (dto: GetMyJobTableDataDto) => {
   );
 };
 
+const DEBOUNCE_TIME_MS = 500;
+
 export function useGetMyJobsData() {
-  const { filterParams } = useMyJobsFilterStore();
-  const { address } = useParams<{ address: string }>();
-  const dto = { ...filterParams, oracle_address: address ?? '' };
+  const {
+    filterParams: { escrow_address, ...filterParams },
+  } = useMyJobsFilterStore();
+  const { address: oracle_address } = useParams<{ address: string }>();
+  const [debouncedEscrowAddress] = useDebounce(
```

Comment:
Doing debounce in query is implicit and violates its responsibility: you might want to use that w/o debouncing, but having it here will cause side effects. So it shouldn't be done here, it should be done in place where we handle the user input.

No need to have "form" component" it can be just regular styled input where you pass `onChange={handleChange}`, and this `handleChange` is something like

```
const handleChange = debounce((event) => {
  queryData(event.target.value);
}, ms);
```

Reply:
I moved debounce to `EscrowAddressSearchForm`. I would preferer to have this input wrapped in form, that gives me convenient validation and error states handling like in all other inputs in app. It that's important for you I can remove that completely, let me know.

Reply:
I also removed changes that were a part of HUM-50